Deploying_Seq2Seq_Model_with_Hybrid_Frontend.ipynb

```diff
@@ -0,0 +1,750 @@
+    "test_decoder_hidden = test_encoder_hidden[:decoder.n_layers]\n",
+    "test_decoder_input = torch.LongTensor(1, 1).random_(0, voc.num_words)\n",
+    "# Trace the model\n",
+    "traced_decoder = torch.jit.trace(decoder, (test_decoder_input, test_decoder_hidden, test_encoder_outputs))\n",
+    "\n",
+    "### Initialize searcher module\n",
+    "scripted_searcher = GreedySearchDecoder(traced_encoder, traced_decoder, decoder.n_layers)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Print Graphs\n",
+    "\n",
+    "Now that our models are in Torch Script form, we can print the graphs of each to ensure that we captured the computational graph appropriately. Since our `scripted_searcher` contains our `traced_encoder` and `traced_decoder`, these graphs will print inline."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#print('searcher graph', scripted_searcher.graph)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Run Evaluation\n",
+    "\n",
+    "Finally, we will run evaluation of the chatbot model using the Torch Script models. If converted correctly, the models will behave exactly as they would in their eager-mode representation.\n",
+    "\n",
+    "By default, we evaluate a few common query sentences. If you want to chat with the bot yourself, uncomment the `evaluateInput` line and give it a spin."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "> hello\n",
+      "Bot: hello .\n",
+      "> what's up?\n",
+      "Bot: i don t know .\n",
+      "> who are you?\n",
+      "Bot: i m the owner .\n",
+      "> where am I?\n",
+      "Bot: i don t know .\n",
+      "> where are you from?\n",
+      "Bot: to the bathroom .\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Evaluate examples\n",
+    "sentences = [\"hello\", \"what's up?\", \"who are you?\", \"where am I?\", \"where are you from?\"]\n",
+    "for s in sentences:\n",
+    "    evaluateExample(s, traced_encoder, traced_decoder, scripted_searcher, voc)\n",
+    "\n",
+    "# Evaluate your input\n",
+    "#evaluateInput(traced_encoder, traced_decoder, scripted_searcher, voc)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Save Model\n",
+    "\n",
+    "Now that we have successfully converted our model to Torch Script, we will serialize it for use in a non-Python deployment environment. To do this, we can simply save our `scripted_searcher` module, as this is the user-facing interface for running inference against the chatbot model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "torch.save(scripted_searcher.state_dict(), \"scripted_chatbot.pth\")"
```

Comment:
This is wrong save. This should be using `scripted_searcher.save("scripted_chatbot.pth")`.

Reply:
Thanks, I'll make this change.